### PR TITLE
Pydantic v2

### PR DESCRIPTION
**Pull request by maparent**
This is the second phase of the work started in the corrections PR, where Pydantic is updated from v1 to v2.
(Sorry for the confusion where the previous PR message was intended for the other branch.)

### CODE REVIEW

**Comment by dcferreira** [comment]
Thanks for the contribution @maparent ! 

Before looking at the code, I don't quite understand why there are 2 PRs; do they both need to be merged, or does one supersede the other? If they both need to be merged, do they need a particular order?

As a side-note, the CI is failing, but I've just merged a fix for it (https://github.com/dcferreira/pylemmy/pull/29). Could I ask you to merge from master, so the CI runs correctly in your code?

**Comment by maparent** [comment]
I separated the PRs because the first one (corrections) should apply in all cases; the second (pydantic_v2) can only applied after the first, but it is a judgement call whether you want to use pydantic v2 directly or on a branch.
Thanks for the ruff upgrade; I'll rebase appropriately.

**Comment by maparent** [comment]
(in case this was not clear, the rebases were done, waiting for workflow approval.)

File: pylemmy/api/comment.py
Comment (dcferreira) [inline pylemmy/api/comment.py]:
```suggestion
```
can simply remove it then! I guess it was in an old version of the API, but no need to keep it

File: pylemmy/api/community.py
Comment (dcferreira) [inline pylemmy/api/community.py]:
```suggestion
```
same as above

File: pylemmy/api/post.py
Comment (dcferreira) [inline pylemmy/api/post.py]:
```suggestion
```

one more

**Review by dcferreira — APPROVED** [review]
Thanks a lot again @maparent ! It seems the linter is complaining, can you also fix that please?

**Comment by dcferreira** [comment]
Oh yeah, it seems this one now has some conflicts unfortunately (after merging #28). Can I ask you to fix those? I'll make a new release once this one is merged

**Comment by dcferreira** [comment]
I've also just invited you to this repo as a collaborator, so you have a bit more permissions. I'm hoping it'll enable you to run the CI by yourself.

File: pylemmy/api/comment.py
Comment (maparent) [reply pylemmy/api/comment.py]: A bit conflicted here; this is found in the official Lemmy api, but setting a default of false (hence using it in the api) causes the test to fail. This will need more study, but removing the comments for now, and adding a few more parameters from the api.

**Comment by maparent** [comment]
Thanks a lot for the invite! If you're ok with that I will continue to add to the API endpoints directly.
The Python 3.7 test fails, I suspect some components are getting old, and Python 3.7 has reached end of life.
I suggest removing it from the test suite, what do you think?

**Comment by dcferreira** [comment]
Yup removing 3.7 is fine. 

> I will continue to add to the API endpoints directly

Not quite sure what you mean; if it's about committing on master, I'd prefer if you still do PRs. Though I think it's not enforced right now, just a preference

**Pull request merged**
